### ISSUE TITLE
Add draft and pending statuses for revisions too

### ISSUE THREAD

**Issue by coreymckrill** [WordPress/revisions-extended#112]
There are times when you want to save some changes you've made to an already-published post, but you don't want to schedule them to live at some point in the future. Maybe the changes are still a "draft" that you'll come back and work more on later. Or maybe they're "pending" and you want someone else to review them before making them live.

Adding these statuses to revisions on the back end isn't that complicated. The trickier part is that since these "update" revisions save the whole post rather than a diff of the changes, you can really only have one at a time. Otherwise you end up with a situation where applying one pending update would invalidate/overwrite a separate pending update. So the UI around "this post already has an update in the works" needs to be robust and also prevent people from inadvertently breaking things.

Either that or we need to investigate whether PHP [xdiff module](https://www.php.net/manual/en/intro.xdiff.php) would be viable to use.

**Comment by tlovett1** [comment]
@coreymckrill I love the idea. I might approach this in a different way. I think you need the ability to create multiple draft revisions for the same post at the same time. Use case is Writer A and B are both working on revisions for the post. The Editor can ultimately pick which revision they want to use. To simplify, if a draft revision were applied all other draft revisions would be invalidated (perhaps a message when the revision is opened up saying it needs to be refreshed from the original post). I wouldn't go down the xdiff route, I'd simply just replace the post wholesale with the revision. Thoughts?

**Comment by coreymckrill** [comment]
Yeah, I think this is what the default behavior would look like without diffing. It would be all about managing expectations within the UI. Like, if you have three different updates for a post (could be any combination of draft/pending/future), and you go to publish one of them, you'd need to show a modal or something that says "here are the other updates that will become obsolete when you publish this." I think you'd still need some kind of diffing functionality if you wanted to provide a reasonable way to "refresh" an obsolete update, though.

**Comment by coreymckrill** [comment]
Side note: I wonder if something like [JSON patch](https://github.com/mikemccabe/json-patch-php) could be used instead of xdiff...
